src/components/data/DataTable.tsx

```diff
@@ -75,6 +97,38 @@ function defaultRenderItem<Row>(r: Row, field: keyof Row): ComponentChildren {
   return r[field] as ComponentChildren;
 }
 
+function calculateNewOrder<T>(newField: T, prevOrder: Order<T>): Order<T> {
+  if (newField !== prevOrder?.field) {
+    return { field: newField, direction: 'ascending' };
+  }
+
+  return prevOrder.direction === 'ascending'
+    ? { field: newField, direction: 'descending' }
+    : null;
+}
+
+type HeaderComponentProps = {
+  onClick?: () => void;
+  children: ComponentChildren;
+  label: string;
+};
+
+function HeaderComponent({ children, onClick, label }: HeaderComponentProps) {
+  const commonClasses = 'flex justify-between items-center';
+  return onClick ? (
+    <Button
+      classes={`${commonClasses} w-full !p-3`}
```

Comment:
Setting `!p-3`is a bit ugly. I considered adding `unstyled`, so that then we can simply add the regular `p-3` class, but it has the side effect that the focus ring styles are not added.